src/bench/pqs/spraylist.cpp

```diff
@@ -55,14 +55,38 @@ void
 spraylist::insert(const uint32_t &k,
                   const uint32_t &v)
 {
-    sl_add(m_q, k, READ_ADD_REM_ELASTIC_TX);
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
+}
+
+void
+spraylist::insert(const size_t &k,
+                  const size_t &v)
+{
+    sl_add_val(m_q, k, v, TRANSACTIONAL);
 }
 
 bool
-spraylist::delete_min(uint32_t &v)
+spraylist::delete_min(size_t &k, size_t &v)
 {
-    const int ret = spray_delete_min(m_q, &v, d);
-    return (ret == 1);
+    unsigned long k_ret;
+    unsigned long v_ret;
+    int ret;
+    do {
+        ret = spray_delete_min_key(m_q, &k_ret, &v_ret, d);
```

Comment:
> It differs from Listing 5 in that failed CAS result in a spurious failure.
Why doesn't the implementation do as the paper? I think that it would make more sense if the implementation retried on CAS failure as the code in the paper does. I think so for the same reasons as I think the retry loop that I added to SprayList's delete_min should be there.

I was wrong in thinking that the KLSM implementation didn't have this kind of spurious failures. I thought the KLSM's spurious failures happened when the issuing thread's local queue and the shared queue both were empty but some other local queue had items, which I could understand as it would be costly to detect this kind of spurious failures.

Anyway, if it is decided to change the KLSM, I think is better that you @schuay do it since you are more familiar with the KLSM code. I will go ahead and do the other changes that you have suggested so we can finish this pull request.